### PR TITLE
TLMBLGIS-17 Change Message to Event and add category

### PR DESCRIPTION
**Pull request by dovek**
DUE: 4/26
@lsloan 
@zqian

### CODE REVIEW

File: mbof/serializers.py [hunk 1]
```diff
@@ -16,9 +16,9 @@ class Meta:
 
 class MessageSerializer(serializers.HyperlinkedModelSerializer):
     class Meta:
-        model = Message
+        model = Event
         fields = ('url', 'messageText', 'latitude', 'longitude', 'altitudeMeters', 'owner', 'postingTime', 'startTime',
```
Comment (zqian) [inline mbof/serializers.py]: Should messageText also be renamed to eventText?

**Comment by zqian** [comment]
Did not see the newly introduced "category" field in the sample dev_data.json file. Should it be added?

File: mbof/models.py [hunk 2]
```diff
@@ -57,11 +57,12 @@ def __str__(self):
 
 
 @python_2_unicode_compatible
-class Message(models.Model):
+class Event(models.Model):
     latitude = models.FloatField()
     longitude = models.FloatField()
     altitudeMeters = models.FloatField()
     messageText = models.CharField(max_length=400)
```
Comment (lsloan) [inline mbof/models.py]: - [x] `messageText` should be `eventText` or something more general.

File: mbof/models.py [hunk 3]
```diff
@@ -93,15 +94,15 @@ def save(self, force_insert=False, force_update=False, using=None,
...
+        return super(Event, self).save(force_insert, force_update, using, update_fields)
 
 
 @python_2_unicode_compatible
 class Vote(models.Model):
     VOTE_PLUS = '+1'
     VOTE_MINUS = '-1'
     VOTE_NONE = '0'
-    message = models.ForeignKey(Message)
+    message = models.ForeignKey(Event)
```
Comment (lsloan) [inline mbof/models.py]: - [x] `message` should be `event`

File: mbof/serializers.py [hunk 4]
```diff
@@ -16,9 +16,9 @@ class Meta:
 
 class MessageSerializer(serializers.HyperlinkedModelSerializer):
```
Comment (lsloan) [inline mbof/serializers.py]: - [x] `MessageSerializer` should be `EventSerializer`

File: mbof/views.py [hunk 5]
```diff
@@ -4,18 +4,18 @@
 from django.shortcuts import render, get_object_or_404
 from rest_framework import viewsets
 
-from .models import Message, User, Vote
+from .models import Event, User, Vote
 from .serializers import MessageSerializer, UserSerializer, VoteSerializer
```
Comment (lsloan) [inline mbof/views.py]: - [x] `MessageSerializer` should be `EventSerializer`

File: mbof/views.py [hunk 6]
```diff
@@ -4,18 +4,18 @@
...
 
 
 def index(request):
     return render(request, 'messages/index.html', {
-        'latestMessageList': Message.objects.order_by('-postingTime'),
+        'latestMessageList': Event.objects.order_by('-postingTime'),
     })
```
Comment (lsloan) [inline mbof/views.py]: - [ ] I believe `index()`, and the following three methods (`detail()`, `results()`, and `vote()`) can be removed.  I think I added these to let me test the API before @gsilver implemented the UI.  Try commenting these out and see what breaks.  Really, all I/O from the Django app should be through the rest API, making these methods (and **_everything**_ in the `templates` directory) aren't needed.

File: mbof/views.py [hunk 7]
```diff
@@ -46,7 +46,7 @@ class MessageViewSet(viewsets.ModelViewSet):
     """
```
Comment (lsloan) [inline mbof/views.py:24]: - [x] `class MessageViewSet` (line not shown in the diffs) should be `class EventViewSet`.

File: mbof/views.py [hunk 8]
```diff
@@ -46,7 +46,7 @@ class MessageViewSet(viewsets.ModelViewSet):
...
     API endpoint that allows users to be viewed or edited.
     """
-    queryset = Message.objects.all().order_by('postingTime')
+    queryset = Event.objects.all().order_by('postingTime')
     serializer_class = MessageSerializer
```
Comment (lsloan) [inline mbof/views.py]: - [ ] `MessageSerializer` should be `EventSerializer`

**Comment by lsloan** [comment]
Missed instances that need to be changed:
- `ocellus/mbof/urls.py`
  - [x] Three occurrences of `messageId` are only used in the response methods I suggested should be deleted
- `ocellus/hacks_mbof/urls.py`
  - [x] Change both mentions of `message` in `router.register(r'messages', views.MessageViewSet)`
- `ocellus/mbof/serializers.py`
  - [x] In `VoteSerializer.fields` , change `message` to `event`
- `mbof/models.py`
  - [x] Change `message` in `Vote.objects.filter(message=self),`
  - [x] Change `messageText` in `return str(self.messageText) + ' (' + self.__class__.__name__ + ': ' + str(self.id) + ')'`
  - [x] Change `message` in `unique_together = ('message', 'voter',)`
  - [x] Change `message` in `self.message) + ' (' + self.__class__.__name__ + ': ' + str(self.id) + ')'`
- `ocellus/mbof/migrations/0001_initial.py`
  - [x] There are several mentions of `message` in this single migration and none of the others.  Rather than fixing the migration, though, you might just want to delete all migrations and start with brand new ones.

**Comment by lsloan** [comment]
- [x] Also keep in mind that many of these changes will change the JSON responses from the API.  That will affect the JS code.  I only searched through the Python code, so you'll need to search the JS code and test it.

**Comment by lsloan** [comment]
PS:  Sorry to take a long time completing this review.

**Comment by lsloan** [comment]
Looks nearly complete.  A few items left from my earlier comments about things that need to be deleted:
- `mbof/templates`
  - [x] Delete this directory and its contents.  They're no longer used.
- `mbof/views.py`
  - [x] Delete these response methods that are no longer being used: `index()`, `detail()`, `results()`, and `vote()`.
- `mbof/urls.py`
  - [x] Three occurrences of `messageId` are only referenced (now as `eventId`) in the `mbof/views.py` response methods I suggested should be deleted.  Just delete the contents of the `urlpatterns` list in this file.

File: mbof/views.py (hunk 6 above)
Comment (lsloan) [inline mbof/views.py]: GitHub says this diff is outdated, but the comment still applies.  😉

**Comment by lsloan** [comment]
:shipit:!

**Pull request merged**